packages/cli/src/UserManagement/routes/users.ts

```diff
@@ -203,4 +201,155 @@ export function addUsersMethods(this: N8nApp): void {
 			return generatePublicUserData(updatedUser);
 		}),
 	);
+
+	this.app.get(
+		`/${this.restEndpoint}/users`,
+		ResponseHelper.send(async (req: Request) => {
+			if ((req.user as User).globalRole.name !== 'owner') {
+				throw new ResponseHelper.ResponseError(
+					'Current user cannot perform this operation',
+					undefined,
+					403,
+				);
+			}
+
+			const users = await Db.collections.User!.find();
+
+			return users.map((user) => generatePublicUserData(user));
+		}),
+	);
+
+	this.app.delete(
+		`/${this.restEndpoint}/users/:id`,
+		ResponseHelper.send(async (req: Request) => {
+			if (
+				(req.user as User).globalRole.name !== 'owner' ||
+				(req.user as User).id === req.params.id
+			) {
+				throw new ResponseHelper.ResponseError(
+					'Current user cannot perform this operation',
+					undefined,
+					403,
+				);
+			}
+
+			const transferId = req.query.transferId as string;
+
+			const searchIds = [req.params.id];
+			if (transferId) {
+				if (transferId === req.params.id) {
+					throw new ResponseHelper.ResponseError(
+						'Removed user and transferred user cannot be the same',
+						undefined,
+						400,
+					);
+				}
+				searchIds.push(transferId);
+			}
+
+			const users = await Db.collections.User!.find({ where: { id: In(searchIds) } });
+			if ((transferId && users.length !== 2) || users.length === 0) {
+				throw new ResponseHelper.ResponseError('Could not find user', undefined, 400);
+			}
+
+			const deletedUser = users.find((user) => user.id === req.params.id) as User;
+
+			if (transferId) {
+				const transferUser = users.find((user) => user.id === transferId) as User;
+				await Db.collections.SharedWorkflow!.update({ user: deletedUser }, { user: transferUser });
+				await Db.collections.SharedCredentials!.update(
+					{ user: deletedUser },
+					{ user: transferUser },
+				);
+			} else {
+				const queryBuilderWorkflows = Db.collections.Workflow!.createQueryBuilder('w');
+				queryBuilderWorkflows.select(['w.id']);
+				queryBuilderWorkflows.innerJoin('w.shared', 'shared');
+				queryBuilderWorkflows.andWhere('shared.userId = :userId', {
+					userId: deletedUser.id,
+				});
+
+				const workflows = await queryBuilderWorkflows.getMany();
+
+				if (workflows.length) {
+					await Db.collections.Workflow!.remove(workflows);
+					await Db.collections.SharedWorkflow!.delete({ user: deletedUser });
+				}
+
+				const queryBuilderCredentials = Db.collections.Credentials!.createQueryBuilder('c');
+				queryBuilderCredentials.innerJoin('c.shared', 'shared');
+				queryBuilderCredentials.andWhere('shared.userId = :userId', {
+					userId: deletedUser.id,
+				});
+
+				queryBuilderCredentials.select(['c.id']);
+
+				const credentials =
+					(await queryBuilderCredentials.getMany()) as unknown as ICredentialsResponse[];
+
+				if (credentials.length) {
+					await Db.collections.Credentials!.remove(credentials);
+					await Db.collections.SharedCredentials!.delete({ user: deletedUser });
+				}
```

Comment:
QueryBuilder is always very clunky and should be last resort imo, suggestion:
```ts
const entries = await Db.collections.SharedWorkflow!.find({ user: deletedUser })`
const sharedWorkflowIds = entries.map(entry => entry.workflowId)
await Db.collections.Workflow!.delete(sharedWorkflowIds)
await Db.collections.SharedWorkflow!.delete({ user: deletedUser })
```
(naming just mocked)

doing the same for the credentials and we have it done in 8 lines of code 👍🏼